### PR TITLE
fix: helm upgrade fix

### LINKED ISSUE #81: Fix GH workflow: Lint and Test Chart - helm upgrade not working as expected
The workflow step _Run helm upgrade_ in workflow _Lint and Test Chart_ is supposed to use the workflow input `upgrade_from` for proper Helm Chart upgrade testing (see example workflow in [TRG 5.09)](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09):

```yaml
- name: Run helm upgrade
  run: |
    helm repo add bitnami https://charts.bitnami.com/bitnami
    helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
    helm install [NAME] tractusx-dev/[CHART] --version ${{ github.event.inputs.upgrade_from || 'x.x.x' }}
    helm dependency update charts/[CHART]
    helm upgrade [NAME] charts/[CHART]
  if: github.event_name != 'pull_request' || steps.list-changed.outputs.changed == 'true'
```
Your workflow is ignoring the workflow input `upgrade_from` and installs the latest available helm chart from our Helm repository and the Helm chart version available from branch, which is used for the workflow run:

https://github.com/eclipse-tractusx/sd-factory/blob/1e60d035e7f012a1d8cc1e0f6a5889018b760ee7/.github/workflows/helm-lint.yaml#L101-L106

There is another GH workflow called _Upgrade Charts_ which is also not set up correctly.

Both workflows do not properly test the helmet upgrade and the [TRG 5.09 - Helmet Test](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09) is not fully met.

#75

### PR DESCRIPTION
**Pull request by adkumar1**
- Fixes #81

### CODE REVIEW

**Comment by adkumar1** [comment]
@adityagajbhiye Please review

File: .github/workflows/helm-lint.yaml [hunk 1]
```diff
@@ -100,7 +100,7 @@ jobs:
         # default value for event_name != workflow_dispatch
       - name: Run helm upgrade
         run: |
-          helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
-          helm install sdf tractusx-dev/sdfactory
-          helm upgrade sdf charts/sdfactory
+          helm repo add tractusx https://eclipse-tractusx.github.io/charts/dev
+          helm install sdf tractusx/sdfactory --version ${{ github.event.inputs.upgrade_from || '2.1.7' }}
+          helm upgrade sdf tractusx/sdfactory
```
Comment (carslen) [inline .github/workflows/helm-lint.yaml:105]: You should use `charts/sdfactory` for upgrade, not the public helm repository. This workflow is designed to test even unreleased helm charts, while opening PRs. You setup would require the have the "upgrade to" helm chart version to be available in public helm chart repo.

File: .github/workflows/helm-upgrade.yaml
Comment (carslen) [inline .github/workflows/helm-upgrade.yaml:1]: Why having 2 workflows for Helm upgrade?

File: charts/sdfactory/Chart.yaml
Comment (carslen) [inline charts/sdfactory/Chart.yaml:1]: Why are you decreasing the helm chart version? There is already a 2.1.8 helm chart released:

```
helm search repo tractusx-dev/sdfactory --versions | head -n5
NAME                  	CHART VERSION	APP VERSION	DESCRIPTION
tractusx-dev/sdfactory	2.1.9        	2.1.8      	Helm Charts for SD Factory application. Self-De...
tractusx-dev/sdfactory	2.1.8        	2.1.7      	Helm Charts for SD Factory application. Self-De...
tractusx-dev/sdfactory	2.1.7        	2.1.6      	Helm Charts for SD Factory application. Self-De...
tractusx-dev/sdfactory	2.1.6        	2.1.6      	Helm Charts for SD Factory application. Self-De...
```

Why at all are you touching the chart when there is no change in the chart?

**Review by carslen — CHANGES_REQUESTED** [review]

**Review by adityagajbhiye9 — COMMENTED** [review]
changes are ok

File: .github/workflows/helm-lint.yaml (hunk 1 above)
Comment (adkumar1) [reply .github/workflows/helm-lint.yaml]: Hi @carslen : I have updated it.

File: .github/workflows/helm-upgrade.yaml
Comment (adkumar1) [reply .github/workflows/helm-upgrade.yaml:1]: Long back it was requested by someone.
Removed it now.

File: charts/sdfactory/Chart.yaml
Comment (adkumar1) [reply charts/sdfactory/Chart.yaml:1]: In tractusX last released version was 2.1.7
https://github.com/eclipse-tractusx/sd-factory/releases

**Review by adkumar1 — COMMENTED** [review]
I have updated it.

**Comment by MehranRoshandel** [comment]
@carslen, can you please help us to merge this PR by today?

File: charts/sdfactory/Chart.yaml
Comment (carslen) [reply charts/sdfactory/Chart.yaml:1]: We're talking about the Helm Chart version, not the app version. In Eclipse-Tractus-X Helm Repository there are already Helm Charts available in version 2.1.8 + 2.1.9. That's why I'm asking why you touched the Helm Chart at all with this PR when there is no other change than version numbers.

This PR is about to fix the GH Workflow for Helm lint/install/upgrade. No need to touch the Helm Chart.

**Review by carslen — APPROVED** [review]

**Pull request merged**